firewood/src/merkle/proof.rs

```diff
@@ -352,31 +353,31 @@ impl<N: AsRef<[u8]> + Send> Proof<N> {
                 }
                 Some(p) => {
                     // Return when reaching the end of the key.
-                    if key_index == chunks.len() {
-                        #[allow(clippy::indexing_slicing)]
-                        (cur_key = &chunks[key_index..]);
+                    if key_nibbles.size_hint().0 == 0 {
                         let mut data = None;
 
                         // Decode the last subproof to get the value.
-                        if let SubProof::Hash(p_hash) = p {
+                        let node_ref = if let SubProof::Hash(p_hash) = p {
                             let proof = proofs_map
                                 .get(&p_hash)
                                 .ok_or(ProofError::ProofNodeMissing)?;
 
-                            chd_ptr = self.decode_node(merkle, cur_key, proof.as_ref(), true)?.0;
+                            chd_ptr = self
+                                .decode_node(merkle, key_nibbles, proof.as_ref(), true)?
+                                .0;
 
                             // Link the child to the parent based on the node type.
-                            match &u_ref.inner() {
+                            #[allow(clippy::indexing_slicing)]
+                            match &node_ref.inner() {
```

Comment:
q: what does the TODO comment mean on the next line? Isn't it obsolete at this point?

Reply:
It's to handle paths in branches. They aren't handled in this PR.